pkg/envoy/xds/metrics.go

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package xds
+
+import (
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+)
+
+const (
+	subsystem            = "xds"
+	typeLabel            = "type"
+	statusLabel          = "status"
+	ackStatusLabelValue  = "ack"
+	nackStatusLabelValue = "nack"
+)
+
+type Metrics interface {
+	IncreaseACK(string)
+	IncreaseNACK(string)
+}
+
+var _ Metrics = (*XDSMetrics)(nil)
+
+type XDSMetrics struct {
+	// ACK is the number of ACK responses from envoy.
+	ACKCount metric.Vec[metric.Counter]
+
+	// NACKCount is the number of NACK errors from envoy.
+	NACKCount metric.Vec[metric.Counter]
+}
+
+func NewXDSMetric() *XDSMetrics {
+	return &XDSMetrics{
+		ACKCount: metric.NewCounterVec(metric.CounterOpts{
```

Comment:
i'm not that familiar with the metrics subsystem, but why do we initialize `ACKCount` but don't do this for `NACKCount`?

edit: ok looks like calling `IncreaseNACK` results in  a panic.

If we have `type` & `status` as labels, shouldn't the metric `ACKCount` be renamed to something else (as it contains ACKs & NACKs - and `NACKCount` deleted completely? And both helperfunctions would use the same metric (just with different status label?)

Reply:
I think something wrong with my IDE, the metric was updated to cilium_xds_events_count{typeURL,status} as per PR description, but seems like i messed up the commit or IDE didn't write to file 😢

Reply:
it seems as this isn't addressed yet. `NACKCount` is still not initialized and an error occurs in case of increasing the nACK metric..

As mentioned: shouldn't we use only one metric for both cases?